### PR TITLE
feat (webapp): show open position

### LINKED ISSUE #1863: Web: open new position

### PR DESCRIPTION
**Pull request by bonomat**
This was a bit of a PITA 🙈

resolves #1863 

image

### CODE REVIEW

File: webapp/.gitignore [hunk 1]
```diff
@@ -110,3 +110,6 @@ Cargo.lock
 *.pdb
 
 # End of https://www.toptal.com/developers/gitignore/api/flutter,dart,rust
+
+# data directory for webapp
+data
```
Comment (luckysori) [inline webapp/.gitignore:115]: I thought we were meaning to share the `data` directory with the root of the project. This is not wrong though.

File: webapp/src/api.rs [hunk 2]
```diff
@@ -131,3 +140,57 @@ pub async fn send_payment(params: Json<Payment>) -> Result<(), AppError> {
...
+            leverage: value
+                .leverage
+                .to_f32()
+                .context("To be able to parse leverage into f32")?,
+            quantity: value
+                .quantity
+                .to_f32()
+                .context("To be able to parse leverage into f32")?,
+            contract_symbol: ContractSymbol::BtcUsd,
+            direction: Direction::Long,
```
Comment (luckysori) [inline webapp/src/api.rs:174]: 📣 I guess you don't want to hard-code this.

File: mobile/native/src/trade/order/mod.rs [hunk 3]
```diff
@@ -13,7 +15,7 @@ mod orderbook_client;
 // When naming this the same as `api_model::order::OrderType` the generated code somehow uses
 // `trade::OrderType` and contains errors, hence different name is used.
 // This is likely a bug in frb.
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
```
Comment (luckysori) [inline mobile/native/src/trade/order/mod.rs:18]: 🔧 I can't see why we need this change. What am I missing?

File: webapp/frontend/lib/wallet/wallet_service.dart [hunk 4]
```diff
@@ -55,11 +45,7 @@ class WalletService {
   }
 
   Future<List<OnChainPayment>> getOnChainPaymentHistory() async {
-    // TODO(holzeis): this should come from the config
-    const port = "3001";
-    const host = "localhost";
-
-    final response = await http.get(Uri.http('$host:$port', '/api/history'));
+    final response = await HttpClientManager.instance.get(Uri(path: '/api/history'));
```
Comment (luckysori) [inline webapp/frontend/lib/wallet/wallet_service.dart:48]: 👍 Nice.

File: webapp/frontend/lib/trade/oder_and_position_table.dart
Comment (luckysori) [inline webapp/frontend/lib/trade/oder_and_position_table.dart:1]: 🔧 Name of file has typo.

**Review by luckysori — COMMENTED** [review]
LGTM in general. Haven't considered how it looks yet, but we can iterate on that.

File: webapp/.gitignore (hunk 1 above)
Comment (bonomat) [reply webapp/.gitignore:115]: Yea, but if you run `cargo run` from the sub folder it will generate this which is not wrong but annoying to have :)

**Review by luckysori — APPROVED** [review]

**Comment by bonomat** [comment]
Played around with the design a bit. A bit better but not perfect yet: 

image

image

**Pull request merged**